tests/acceptance/features/lib/UserImpersonatePage.php

```diff
@@ -0,0 +1,89 @@
+<?php declare(strict_types=1);
+
+/**
+ * ownCloud
+ *
+ * @author Paurakh Sharma Humagain <paurakh@jankaritech.com>
+ * @copyright Copyright (c) 2018 Paurakh Sharma Humagain paurakh@jankaritech.com
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License,
+ * as published by the Free Software Foundation;
+ * either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace Page;
+
+use Behat\Mink\Session;
+use Exception;
+
+/**
+ * Admin Apps Settings page.
+ */
+class UserImpersonatePage extends UsersPage {
+	/**
+	 *
+	 * @var string $path
+	 */
+	protected $path = '/index.php/settings/users';
+	protected $impersonateXpath = './/a[@class="action permanent impersonate"]';
+	protected $userXpath = '//span[@id="expandDisplayName"]';
+	protected $impersonateNotificationXpath = '//div[@id="notification-container"]//div[@id="impersonate-notification"]/a';
+
+	/**
+	 * @param Session $session
+	 * @param string $user
+	 *
+	 * @throws Exception
+	 */
+	public function impersonateUser(Session $session, string $user):void {
+		$userTr= $this->findUserInTable($user);
```

Comment:
```suggestion
		$userFromTable= $this->findUserInTable($user);
```
or any nice words...